templates/function-template-nodejs/components/logger.js

```diff
@@ -0,0 +1,106 @@
+/**
+ Nodejs Template Project for Azure
+ @module: logger.js
+ @description: a simple logging module for nodejs
+ @author:
+ @version: 1.0
+ **/
+
+
+/*
+USAGE
+
+    logger = require('./components/logger.js')
+
+    logger.init(event, context); // by default logging level is info
+
+    logger.warn('The following waring message will be logged');
+    logger.verbose('The following waring message will not be logged')
+
+
+    // Sample logging messages.
+    logger.error('Runtime errors or unexpected conditions.');
+    logger.warn('Runtime situations that are undesirable or unexpected, but not necessarily "wrong".');
+    logger.info('Interesting runtime events (Eg. connection established, data fetched etc).');
+    logger.verbose('Generally speaking, most lines logged by your application should be written as verbose.');
+
+    // Alternate usage
+    logger.log('error', 'message');
+    logger.log('info', 'message');
+
+*/
+module.exports = function () {
+  const logLevels = {
+    error: 4,
+    warn: 3,
+    info: 2,
+    verbose: 1
+  };
+
+  const config = {
+    curLogLevel: 'info',
+    requestDetails: ''
+  };
+  let context;
+  // set logLevel, RequestDetails
+  const init = (event, context) => setLevel('info', context);
+
+  // set current logLevel; Only logs which are above the curLogLevel will be logged;
+  const setLevel = (level, context) => {
+    // LOG_LEVEL is 'info' by default
+    this.context = context;
+
+    if (level && logLevels[level]) {
+      // If LOG_LEVEL if explicitly specified , set it as the curLogLevel
+      config.curLogLevel = level;
+      return level;
+    } else {
+      // Get LOG_LEVEL from the environment variables (if defined)
+      try {
+        level = process.env.LOG_LEVEL;
```

Comment:
Same as other templates. Will this work in azure environment?

Reply:
For what i know from doc:
https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-node
it looks like it is the same process.env.  But this has a carva, we are using azure log instead of console log, and azure log default level (similar as java slf4j properties file) is controlled in the host.json.  And host.json has special syntax for logging.  

`
module.exports = function (context, myTimer) {
    var timeStamp = new Date().toISOString();

    context.log('Node.js timer trigger function ran!', timeStamp);
    context.log(GetEnvironmentVariable("AzureWebJobsStorage"));
    context.log(GetEnvironmentVariable("WEBSITE_SITE_NAME"));

    context.done();
};

function GetEnvironmentVariable(name)
{
    return name + ": " + process.env[name];
}
`